src/services/CatalogService.ts

```diff
@@ -31,4 +44,60 @@ export const catalogService: ICatalogService = {
 
     await events.createEvent(eserviceSeedToCreateEvent(eserviceSeed));
   },
+  updateEService: async (
+    eServiceId: string,
+    eservicesSeed: ApiEServiceSeed
+  ): Promise<void> => {
+    const organizationId = await readModelGateway.getOrganizationID();
+    const eservice = await readModelGateway.getEServiceById(eServiceId);
+
+    if (eservice === undefined) {
+      throw eServiceNotFound(eServiceId);
+    }
+
+    if (eservice.producerId !== organizationId) {
+      throw operationForbidden;
+    }
+
+    if (
+      !(
+        eservice.descriptors.length === 0 ||
+        (eservice.descriptors.length === 1 &&
+          eservice.descriptors[0].state === "DRAFT")
+      )
+    ) {
+      throw eServiceCannotBeUpdated(eServiceId);
+    }
+
+    const eserviceSeed = convertToClientEServiceSeed(
+      eservicesSeed,
+      organizationId
+    );
+
+    await events.createEvent({
+      streamId: eServiceId,
+      version: await readModelGateway.getVersion(eServiceId),
+      type: "EServiceUpdated",
+      data: eserviceSeed,
+    });
+  },
+  deleteEService: async (eServiceId: string): Promise<void> => {
+    const organizationId = await readModelGateway.getOrganizationID();
+    const eservice = await readModelGateway.getEServiceById(eServiceId);
+
+    if (eservice === undefined) {
+      throw eServiceNotFound(eServiceId);
+    }
+
+    if (eservice.producerId !== organizationId) {
+      throw operationForbidden;
+    }
+
+    await events.createEvent({
```

Comment:
I think there's also a check on whether it can be deleted: https://github.com/pagopa/interop-be-catalog-management/blob/1.0.x/src/main/scala/it/pagopa/interop/catalogmanagement/api/impl/EServiceApiServiceImpl.scala/#L582